### ISSUE TITLE
Include py.typed file

### ISSUE THREAD

**Issue by canassa** [johnthagen/python-blueprint#24]
Hi!

This is a great project idea!

I noticed that you didn't include a py.typed in your project. It's very useful as it tells that your project is typed

More information here:

https://www.python.org/dev/peps/pep-0561/

**Comment by johnthagen** [comment]
Thanks for the heads up! Wow, I have never even heard of this file (which is probably not a good sign for its adoption) but I will do some research and likely add it to the project.

**Comment by canassa** [comment]
Yeah, It's obscure. I found out about while I was researching why I wasn't getting type checking from a library while the library had type annotations. I added the py.typed file and I instantly got the type checking working again.

It's a bit tricky to get that file into the package since it's not `.py` file. I had to use the `package_data` option to include it, this is my `setup.cfg` as an example:

```
[metadata]
name = super_library
author = ...
author_email = ...
version = 2.15.0
url = ...
long_description = file: README.md
long_description_content_type = text/markdown
classifiers =
    Development Status :: 5 - Production/Stable
    Programming Language :: Python :: 3
    Programming Language :: Python :: 3.8
    License :: OSI Approved :: MIT License

[options]
python_requires = >= 3.6.6
install_requires =
    requests
    pydantic>=1.0
    typing_extensions
setup_requires =
    setuptools >= 38.6
    pip >= 10
package_dir=
    =src
packages=find:

[options.packages.find]
where=src

[options.package_data]
super_library = py.typed
```

Btw, you can also move most of the stuff from the `setup.py` file into the `setup.cfg`, `setuptools` supports that.

**Comment by johnthagen** [comment]
For reference, `mypy` documentation on this: https://mypy.readthedocs.io/en/stable/installed_packages.html#making-pep-561-compatible-packages

**Comment by johnthagen** [comment]
> Btw, you can also move most of the stuff from the setup.py file into the setup.cfg, setuptools supports that.

Yeah, since I do a little logic in `setup.py` (i.e. reading `requirements.txt` so as to avoid duplicating the requirements) I so far haven't migrated to `setup.cfg` as I don't think I'd be able to do it fully yet.

In the future, I plan to move to Poetry (#16) as I feel like that is the long term solution.

**Comment by canassa** [comment]
Be careful with Poetry and similar tools as their bus factor is too low IMO. If anything breaks you will be depending on one guy to fix while pip and setuptools and maintained by the PSF.

I personally had a problem with it where we migrate our pipelines to use Poetry but then we found a specific use case that Poetry didn't support correct (installing dependencies from tarballs). I submitted a PR to Poetry but it still took months to get it fixed. Also, the main selling point of Poetry, which is its dependency resolution system, is already supported on pip now with a custom flag.
